File: Unit_tests_python/EX-06-display_text_file_gui/Bard.py
import tkinter as tk
import os

def display_text_file(file_path):
   """Reads, parses, sorts, and displays contents of a .txt file."""

   try:
       with open(file_path, "r") as file:
           content = file.readlines()

       parsed_data = []
       for line in content:
           name, value = line.strip().split(":")
           parsed_data.append({"name": name, "value": value})

       parsed_data.sort(key=lambda item: item["value"], reverse=True)

       temp_file_path = "test.txt"
       with open(temp_file_path, "w") as temp_file:
           for item in parsed_data:
               temp_file.write(f"{item['name']}: {item['value']}\n")

       # Display contents in GUI window
       root = tk.Tk()
       text_box = tk.Text(root)
       text_box.pack()
       with open(temp_file_path, "r") as temp_file:
           text_box.insert("end", temp_file.read())
       root.mainloop()

       return True, os.path.abspath(temp_file_path)

   except FileNotFoundError:
       print("Error: The specified file path does not exist.")
       return False, None
   except Exception as e:
       print(f"An unexpected error occurred: {e}")
       return False, None

File: Unit_tests_lua/EX-06-display_text_file_gui/Bard.lua
function display_text_file(file_path)
    local success, data = pcall(io.open, file_path)
    if not success then
      return false, nil, "Error opening file: " .. data
    end
   
    local lines = data:read("*all"):split("\n")
    local pairs = {}
   
    for _, line in ipairs(lines) do
      local name, value = line:match("([^:]+):([^:]+)")
      if name and value then
        table.insert(pairs, { name = name, value = tonumber(value) })
      end
    end
   
    table.sort(pairs, function(a, b) return a.value > b.value end)
   
    local temp_file = io.open("test.txt", "w")
    if not temp_file then
      data:close()
      return false, nil, "Error creating temporary file."
    end
   
    for _, pair in ipairs(pairs) do
      temp_file:write(pair.name .. ": " .. tostring(pair.value) .. "\n")
    end
   
    temp_file:close()
    data:close()
   
    -- Display the contents of the temporary file in a window GUI (replace with appropriate GUI calls)
    -- local gui_library = require("your_gui_library")
    -- gui_library.show_text_file("test.txt")
   
    return true, "test.txt"
   end   